controllers/machineset.go

```diff
@@ -503,8 +519,25 @@ func (r *AppWrapperReconciler) addLabelToNode(ctx context.Context, aw *arbv1.App
 	return nil
 }
 
-func (r *AppWrapperReconciler) removeLabelFromMachine(ctx context.Context, aw *arbv1.AppWrapper, machineName string) error {
-	labelPatch := fmt.Sprintf(`[{"op":"remove","path":"/metadata/labels/%s","value":"%s" }]`, aw.Name, aw.Name)
+func (r *AppWrapperReconciler) patchNodeLabels(ctx context.Context, oldAw *arbv1.AppWrapper, newAw *arbv1.AppWrapper, nodeName string) error {
+	// Remove Node label
+	labelPatch := fmt.Sprintf(`[{"op":"remove","path":"/metadata/labels/%s","value":"%s" }]`, oldAw.Name, oldAw.Name)
+	_, err := r.kubeClient.CoreV1().Nodes().Patch(ctx, nodeName, types.JSONPatchType, []byte(labelPatch), metav1.PatchOptions{})
+	if err != nil {
+		klog.Infof("Error deleted label to machines %v", err)
+		return err
+	}
+
+	// Add Node label
+	if err := r.addLabelToNode(ctx, newAw, nodeName); err != nil {
```

Comment:
The two patches can be collapsed into a single patch request. The `labelPatch` can be an array of operations, so the one from `addLabelToNode` can be added to the one here, and the `addLabelToNode` method removed.